### PR TITLE
Upgrade to kubernetes 1.0.3

### PR DESCRIPTION
**Pull request by kikov79**
Hi,

I've upgrade your code to make it work with Kubernetes 1.0.3.

* I had to adapt the specs to the v1 format.
* I fix several issues related to the name of the environment variables
* I use the token from /var/run/secrets/kubernetes.io/serviceaccount/token as it's the recommended way to access API within a pod.

You need the AdmissionController option enabled in your apiserver.

Thanks for you great work!

BR,

### CODE REVIEW

**Comment by cvallance** [comment]
Awesome! Thanks @kikov79.

Could I ask for 1 favour to add to this pull request? Will add comment to the code.

File: example/mongo-test-controller.yaml [hunk 1]
```diff
@@ -1,32 +1,35 @@
...
+            - rs0
+            - "--smallfiles"
+            - "--noprealloc"
+          ports:
+            - containerPort: 27017
+        - name: mongo-sidecar
+          image: leportlabs/mongo-k8s-sidecar
+          env:
+            - name: MONGO_SIDECAR_POD_LABELS
+              value: "name:mongo,environment:test"
```
Comment (cvallance) [inline example/mongo-test-controller.yaml]: Lets change the delimiter to be a '=' so it's more inline with kubectl. I.e. `value: "name=mongo,environment=test"`

File: src/lib/config.js [hunk 2]
```diff
@@ -7,10 +7,9 @@ var getMongoPodLabelCollection = function() {
   if (!podLabels) {
     return false;
   }
-
-  var labels = process.env.LL_MONGO_POD_LABELS.split(',');
+  var labels = process.env.MONGO_SIDECAR_POD_LABELS.split(',');
   for (var i in labels) {
-    var keyAndValue = labels[i].split('=');
+    var keyAndValue = labels[i].split(':');
```
Comment (cvallance) [inline src/lib/config.js]: Change to reflect request above.

**Comment by kikov79** [comment]
Done.

**Comment by cvallance** [comment]
Thanks!

**Pull request merged**